src/hcpp-runtime.c

```diff
@@ -246,11 +265,21 @@ void hcpp_entrypoint() {
         fprintf(stderr, "Error in pthread_attr_init\n");
         exit(3);
     }
-    int starting_worker = 1;
-#ifdef HCPP_COMM_WORKER
-    starting_worker = 2;
+
+    for (int i = 1; i < hcpp_context->nworkers; i++) {
+#ifdef HC_COMM_WORKER
+        /*
+         * If running with a thread dedicated to network communication (e.g. through
+         * UPC, MPI, OpenSHMEM) then skip creating that thread as a compute worker.
+         */
+        assert(communication_worker_id > 0);
```

Comment:
It simplifies the code. Thread 0 is the main entry thread, spawned by the system. Thread 1 is the communication thread. Thread 2 is the GPU thread. I don't see anywhere in your code where you assume that Thread 0 is the communication thread (and if that is the case, it's probably something that should be factored out anyway). Again, can you explain how this breaks anything?